src/vs/workbench/parts/debug/common/debug.ts

```diff
@@ -55,23 +55,30 @@ export interface IThread extends ITreeElement {
 	threadId: number;
 	name: string;
 	stoppedDetails: IRawStoppedDetails;
+
+	/**
+	 * Queries the debug adapter for the callstack and returns a promise with
+	 * the stack frames of the callstack.
+	 * If the thread is not stopped, it returns a promise to an empty array.
+	 */
 	getCallStack(debugService: IDebugService): TPromise<IStackFrame[]>;
-	clearCallStack(): void;
 
 	/**
-	 * Indicates whether the call stack for this thread can be obtained from
-	 * the debug adapter.
+	 * Gets the callstack if it has already been received from the debug
+	 * adapter, otherwise it returns undefined.
 	 */
-	callStackObtainable: boolean;
+	getCachedCallStack(): IStackFrame[];
 
 	/**
-	 * Indicates whether the call stack for this thread is available because it
-	 * has already been obtained from the debug adapter.
-	 * This property is used to determine whether the call stack is available
-	 * without triggering a call to the debug adapter.
+	 * Invalidates the callstack cache
 	 */
-	callStackAvailable: boolean;
+	clearCallStack(): void;
 
+	/**
+	 * Indicates whether this thread is stopped. The callstack for stopped
+	 * threads can be retrieved from the debug adapter.
+	 */
+	stopped: boolean;
```

Comment:
`stopped` doesn't necessarily need to be exposed. We could have two methods `setRunning` and `setStopped` that would set `stopped` to false and `true`, respectively.

I left it exposed so that the tests can check it. I'm fine removing `stopped` from the interface and only have setters. Let me know what you think.